contrib/jemalloc-cmake/CMakeLists.txt

```diff
@@ -22,7 +22,12 @@ if (ENABLE_JEMALLOC)
             #
             # By enabling percpu_arena number of arenas limited to number of CPUs and hence
             # this problem should go away.
-            set (JEMALLOC_CONFIG_MALLOC_CONF "percpu_arena:percpu,oversize_threshold:0")
+            #
+            # muzzy_decay_ms -- use MADV_FREE when available on newer Linuxes, to
+            # avoid spurious latencies and additional work associated with
+            # MADV_DONTNEED. See
+            # https://github.com/ClickHouse/ClickHouse/issues/11121 for motivation.
+            set (JEMALLOC_CONFIG_MALLOC_CONF "percpu_arena:percpu,oversize_threshold:0,muzzy_decay_ms:10000")
```

Comment:
Maybe worth enable it for other OSes too, just in case? (although not sure that it has analogs)

Reply:
I think MacOS has some similar madvise call for this, don't know about other systems. Let's enable it.